src/Avalonia.Input/TouchDevice.cs

```diff
@@ -3,22 +3,27 @@
 using System.Linq;
 using Avalonia.Input.Raw;
 using Avalonia.Platform;
+using Avalonia.VisualTree;
 
 namespace Avalonia.Input
 {
     /// <summary>
     /// Handles raw touch events
+    /// </summary>
     /// <remarks>
     /// This class is supposed to be used on per-toplevel basis, don't use a shared one
     /// </remarks>
-    /// </summary>
-    public class TouchDevice : IInputDevice, IDisposable
+    public class TouchDevice : IPointerDevice, IDisposable
     {
         private readonly Dictionary<long, Pointer> _pointers = new Dictionary<long, Pointer>();
         private bool _disposed;
         private int _clickCount;
         private Rect _lastClickRect;
         private ulong _lastClickTime;
+        private Pointer? _lastPointer;
+
+        IInputElement? IPointerDevice.Captured => _lastPointer?.Captured;
```

Comment:
IPointerDevice requires to implement these members. 
I plan to create another PR after this one to remove obsolete members from pointer-related code (at least most of them).